### ISSUE TITLE
std/async: Uncaught exception in pooledMap

### ISSUE THREAD

**Issue by wperron** [denoland/deno#9070]
When an exception occurs in an async callback in `pooledMap`, the exception is caught by surrounding try/catch but it _also_ bubbles into an uncaught Promise Exception and crashes the process.

Example stack trace:
```text
2021-01-05T09:57:47.331+01:00	error: Uncaught (in promise) Http: error sending request for url (https://REDACTED.s3.us-east-1.amazonaws.com/aws_sdk/versions/v3.0.0.1/raw/client-codebuild/pagination/ListBuildBatchesForProjectPaginator.ts): connection closed before message completed
2021-01-05T09:57:47.331+01:00	at processResponse (deno:core/core.js:223:11)
2021-01-05T09:57:47.331+01:00	at Object.jsonOpAsync (deno:core/core.js:240:12)
2021-01-05T09:57:47.331+01:00	at async fetch (deno:op_crates/fetch/26_fetch.js:1278:29)
2021-01-05T09:57:47.331+01:00	at async S3Bucket.putObject (<https://deno.land/x/s3@0.2.0/src/bucket.ts>:352:1)
2021-01-05T09:57:47.331+01:00	at async uploadVersionRaw (<file:///var/task/utils/storage.ts>:82:1)
2021-01-05T09:57:47.331+01:00	at async <file:///var/task/api/async/publish.ts>:170:1
```

I also made a simple gist to reproduce the issue: https://gist.github.com/wperron/1066ad43a77a3cd7aeb3bdecfa44d9d1

cc @lucacasonato

**Comment by wperron** [comment]
I'll investigate – from discussion I had offline with @lucacasonato we'll probably need to rewrite the implementation of `pooledMap`

**Comment by coastalwhite** [comment]
I looked at it shortly and noticed something which I found interesting.

If we add even a little bit of delay to your `cb` function resolving the error is now entirely uncaught. Maybe, this would mean it is a race condition. Maybe it can hint at a race condition. Maybe this has some very easy explanation, but I found it quite weird behaviour.

The code for simulating both types be found in here:
https://gist.github.com/coastalwhite/6d000ee5bfe3628dd57528310328a49c
